### PR TITLE
Use a separated database table to store the data version

### LINKED ISSUE #12747: The data migration runs for every installation
This is caused by that the data schema shares a same table `schema_migrations` with database schema, and when database migration runs, it empties the `schema_migrations` table and inserts a new record every time, so the data schema information lost.

### PR DESCRIPTION
**Pull request by ywk253100**
Use a separated database table to store the data version.
Fixes #12747

Signed-off-by: Wenkai Yin <yinw@vmware.com>

### CODE REVIEW

File: src/migration/artifact.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ func abstractArtData(ctx context.Context) error {
 			continue
 		}
 		for _, repo := range repos {
-			log.Debugf("abstracting artifact metadata under repository %s ....", repo.Name)
+			log.Infof("abstracting artifact metadata under repository %s ....", repo.Name)
```
Comment (reasonerjt) [inline src/migration/artifact.go:45]: Is it possible to add another log message like "listing artifacts for migration..."?
The list may take a long time.

File: make/migrations/postgresql/0040_2.1.0_schema.up.sql [hunk 2]
```diff
@@ -116,3 +116,15 @@ ALTER TABLE schedule DROP COLUMN IF EXISTS status;
 UPDATE registry SET type = 'quay' WHERE type = 'quay-io';
 
 ALTER TABLE artifact ADD COLUMN icon varchar(255);
+
+CREATE TABLE IF NOT EXISTS data_migrations (
```
Comment (reasonerjt) [inline make/migrations/postgresql/0040_2.1.0_schema.up.sql:120]: Can we add constraint to make sure there's only one record?

File: src/migration/artifact.go (hunk 1 above)
Comment (ywk253100) [reply src/migration/artifact.go:45]: The artifact list is read from database

**Comment by ywk253100** [comment]
Will submit another PR to do the update according to the comment

**Pull request merged**